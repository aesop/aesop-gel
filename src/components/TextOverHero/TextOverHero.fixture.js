export default {
  copy:
    'Offering an exceptionally mild yet thorough cleanse, Gentle Facial Cleansing Milk removes surface impurities without stripping moisture, making it especially beneficial for dry or sensitive skin.',
  cta: {
    id: 'button-id',
    openInANewWindow: true,
    url: 'http://google.com',
    text: 'Read more',
    type: 'Absolute',
    style: 'Internal Button Link',
  },
  id: 'ImageHeader-old-component',
  heading: 'Gentle Facial Cleansing Milk',
  eyebrow: 'Image Header old component',
  image: {
    altText: 'Image of Milk Cleansing Campaign Aesop 2019',
    large:
      'https://www.aesop.com/medias/GelCleanser-640x640.png?context=bWFzdGVyfGltYWdlc3w1MzE5NTl8aW1hZ2UvcG5nfGltYWdlcy9oOTUvaGNkLzg4MDc3OTc4NTAxNDIucG5nfGM3NTNhYjdkMWNiOTRhNWZhNDA0NWExZWJiODMxN2VhNzEyY2ZkNGUyYzc4ZWVkZDRjYTIzMzZmZTRlOGYxNjM',
    medium:
      'https://www.aesop.com/medias/GelCleanser-640x640.png?context=bWFzdGVyfGltYWdlc3w1MzE5NTl8aW1hZ2UvcG5nfGltYWdlcy9oOTUvaGNkLzg4MDc3OTc4NTAxNDIucG5nfGM3NTNhYjdkMWNiOTRhNWZhNDA0NWExZWJiODMxN2VhNzEyY2ZkNGUyYzc4ZWVkZDRjYTIzMzZmZTRlOGYxNjM',
    small:
      'https://www.aesop.com/medias/GelCleanser-640x640.png?context=bWFzdGVyfGltYWdlc3w1MzE5NTl8aW1hZ2UvcG5nfGltYWdlcy9oOTUvaGNkLzg4MDc3OTc4NTAxNDIucG5nfGM3NTNhYjdkMWNiOTRhNWZhNDA0NWExZWJiODMxN2VhNzEyY2ZkNGUyYzc4ZWVkZDRjYTIzMzZmZTRlOGYxNjM',
  },
};
