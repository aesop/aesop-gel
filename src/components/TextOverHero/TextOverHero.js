import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import useWindowHasResized from '~/customHooks/useWindowHasResized';
import { ascertainIsMobile } from '~/utils/viewports';
import BodyCopy from '~/components/BodyCopy';
import Image from '~/components/Image';
import styles from './TextOverHero.module.css';

const TextOverHero = ({
  className,
  contentThemeOnMobile,
  copy,
  cta,
  eyebrow,
  heading,
  id,
  image,
  subHeading,
  theme,
}) => {
  useWindowHasResized();

  const isMobile = ascertainIsMobile();
  const classSet = cx(styles.base, styles[theme], className);

  return (
    <section className={classSet}>
      <Image
        altText={image.altText}
        large={image.large}
        medium={image.medium}
        small={image.small}
      />
      <div className={styles.contentWrapper}>
        <div className={styles.content}>
          <BodyCopy
            copy={copy}
            cta={cta}
            eyebrow={eyebrow}
            heading={heading}
            id={id}
            subHeading={subHeading}
            theme={isMobile ? contentThemeOnMobile : theme}
          />
        </div>
      </div>
    </section>
  );
};

TextOverHero.propTypes = {
  className: PropTypes.string,
  contentThemeOnMobile: PropTypes.oneOf(['dark', 'light']),
  copy: PropTypes.string,
  cta: PropTypes.object,
  eyebrow: PropTypes.string,
  heading: PropTypes.string.isRequired,
  id: PropTypes.string,
  image: PropTypes.object.isRequired,
  subHeading: PropTypes.string,
  theme: PropTypes.oneOf(['dark', 'light']),
};

TextOverHero.defaultProps = {
  className: undefined,
  contentThemeOnMobile: 'dark',
  copy: undefined,
  cta: undefined,
  eyebrow: undefined,
  heading: undefined,
  id: undefined,
  image: undefined,
  subHeading: undefined,
  theme: 'dark',
};

export default TextOverHero;
