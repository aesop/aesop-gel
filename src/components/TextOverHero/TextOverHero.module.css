@import 'styles/variables/index';
@import 'styles/mixins/index';

.base {
  @extend %layout-full-bleed-negative-margin;

  position: relative;
  margin-bottom: var(--layout-md-spacing);

  @media (--viewport-lg) {
    margin-bottom: calc(var(--layout-lg-spacing) * 1.5);
  }
}

.noTopMargin {
  margin-top: 0;
}

figure {
  img,
  video {
    display: block;
  }
}

.contentWrapper {
  position: relative;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 100%;
  padding: 40px 0;
  background-color: var(--color-white-dark);

  @media (--viewport-md) {
    position: absolute;
    top: 0;
    left: 0;
    padding: 0;
    background-color: transparent;
  }
}

.content {
  @media (--viewport-md) {
    @mixin grid-pct 7, 12;

    padding-top:
      calc(
        (var(--layout-logo-offset-md) + var(--layout-menu-height) - 10px)
      );
    margin-left: calc((100% / 12) * 4);
  }

  @media (--viewport-lg) {
    @mixin grid-pct 5, 12;

    max-width: 590px;
    padding-top:
      calc(
        (var(--layout-logo-offset-lg) + var(--layout-menu-height) - 5px)
      );
    margin-left: calc((100% / 12) * 3);
  }
}

.primaryTitle,
.secondayTitle {
  margin: 0;
}
