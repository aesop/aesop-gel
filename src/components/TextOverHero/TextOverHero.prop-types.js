import PropTypes from 'prop-types';

export const VIDEO_PROP_TYPE_SHAPE = PropTypes.object;
export const HYPERLINK_PROP_TYPE_SHAPE = PropTypes.object;

export const PROP_TYPES = {
  className: PropTypes.string,
  copy: PropTypes.string,
  cta: HYPERLINK_PROP_TYPE_SHAPE,
  eyebrow: PropTypes.string,
  hasLightText: PropTypes.bool,
  id: PropTypes.string,
  noTopMargin: PropTypes.bool,
  subtitle: PropTypes.string,
  title: PropTypes.string.isRequired,
  video: VIDEO_PROP_TYPE_SHAPE.isRequired,
};

export default PROP_TYPES;
