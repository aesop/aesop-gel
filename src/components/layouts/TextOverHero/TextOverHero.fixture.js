export default {
  copy:
    'Offering an exceptionally mild yet thorough cleanse, Gentle Facial Cleansing Milk removes surface impurities without stripping moisture, making it especially beneficial for dry or sensitive skin.',
  cta: {
    id: 'button-id',
    openInANewWindow: true,
    url: 'http://google.com',
    text: 'Read more',
    type: 'Absolute',
    style: 'Internal Button Link',
  },
  id: 'ImageHeader-old-component',
  title: 'Gentle Facial Cleansing Milk',
  video: {
    autoplay: true,
    id: 'TextOverHeroImage',
    loop: true,
    fallbackImage: {
      altText: 'Image of one of the offices in Tokyo',
      fullBleedImage: true,
      id: 'fullBleed-internal',
      large:
        'https://release.aesop-web-ui.aesopdigital.io/images/full-bleed/large/mc-image-header-2.jpg 2x',
      medium:
        'https://release.aesop-web-ui.aesopdigital.io/images/full-bleed/medium/mc-image-header-2.jpg 2x',
      small:
        'https://release.aesop-web-ui.aesopdigital.io/images/full-bleed/small/mc-image-header-2.jpg 2x',
    },
  },
};
