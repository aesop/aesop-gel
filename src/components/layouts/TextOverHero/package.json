{
  "name": "TextOverHero",
  "version": "0.0.0",
  "main": "TextOverHero.js"
}
