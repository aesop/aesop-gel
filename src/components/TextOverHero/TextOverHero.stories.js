import React from 'react';
import { storiesOf } from '@storybook/react';
import { boolean, text, select } from '@storybook/addon-knobs';
import TextOverHero from './TextOverHero';
import TextOverHeroFixture from './TextOverHero.fixture';

console.log('TextOverHeroFixture', TextOverHeroFixture);

const styleOptions = [
  'External Button Link',
  'External Text Link',
  'Internal Button Link',
  'Internal Text Link',
  'No Icon Link',
];

storiesOf('Components/TextOverHero', module).add('Base component', () => (
  <TextOverHero
    contentThemeOnMobile={select(
      'contentThemeOnMobile',
      ['dark', 'light'],
      'dark',
    )}
    copy={text(
      'copy',
      'Offering an exceptionally mild yet thorough cleanse, Gentle Facial Cleansing Milk removes surface impurities without stripping moisture, making it especially beneficial for dry or sensitive skin.',
    )}
    cta={{
      url: text('cta.url', 'http://aesop.com'),
      text: text('cta.text', 'Read more'),
      style: select('cta.style', styleOptions, 'Internal Button Link'),
    }}
    eyebrow={text('eyebrow', 'Mild yet thorough cleanse')}
    heading={text('heading', 'Gentle Facial Cleansing Milk')}
    id={text('id', 'ImageHeader-old-component')}
    image={TextOverHeroFixture.image}
    theme={select('theme', ['dark', 'light'], 'light')}
  />
));

storiesOf('Components/TextOverHero', module).add('With Secondary CTA', () => (
  <TextOverHero
    copy="Offering an exceptionally mild yet thorough cleanse, Gentle Facial Cleansing Milk removes surface impurities without stripping moisture, making it especially beneficial for dry or sensitive skin."
    cta={{
      text: 'About page',
      url: 'http://aesop.com',
      style: 'External Button Link',
    }}
    heading="Gentle Facial Cleansing Milk"
    image={TextOverHeroFixture.image}
    secondaryCta={{
      text: 'Read more',
      url: 'http://aesop.com',
      style: 'External Button Link',
    }}
    theme="light"
  />
));

storiesOf('Components/TextOverHero', module).add('Dark Theme', () => (
  <TextOverHero
    copy="Offering an exceptionally mild yet thorough cleanse, Gentle Facial Cleansing Milk removes surface impurities without stripping moisture, making it especially beneficial for dry or sensitive skin."
    cta={{
      url: 'http://aesop.com',
      text: 'Read more',
      style: 'Internal Button Link',
    }}
    eyebrow="Mild yet thorough cleanse"
    heading="Gentle Facial Cleansing Milk"
    image={TextOverHeroFixture.lightImage}
    secondaryCta={{
      text: 'Read more',
      url: 'http://aesop.com',
      style: 'External Button Link',
    }}
    theme="dark"
  />
));

storiesOf('Components/TextOverHero', module).add('With Sub Nav', () => (
  <TextOverHero
    heading="Gentle Facial Cleansing Milk"
    image={TextOverHeroFixture.image}
    subNavLinks={TextOverHeroFixture.subNavLinks}
    theme="light"
  />
));
