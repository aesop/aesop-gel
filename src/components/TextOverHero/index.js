import TextOverHero from './TextOverHero';

export default TextOverHero;
