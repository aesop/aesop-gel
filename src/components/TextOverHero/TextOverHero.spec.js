import React from 'react';
import renderer from 'react-test-renderer';
import TextOverHero from './TextOverHero';
import TextOverHeroFixture from './TextOverHero.fixture';

describe('<TextOverHero />', () => {
  it('should be defined', () => {
    expect(TextOverHero).toBeDefined();
  });

  it('renders base component correctly', () => {
    const tree = renderer
      .create(
        <TextOverHero
          copy={TextOverHeroFixture.copy}
          cta={TextOverHeroFixture.cta}
          heading={TextOverHeroFixture.heading}
          id={TextOverHeroFixture.id}
          image={TextOverHeroFixture.image}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
