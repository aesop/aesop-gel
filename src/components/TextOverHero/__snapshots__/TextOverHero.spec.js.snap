// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextOverHero /> renders base component correctly 1`] = `
<section
  className="base dark"
>
  <picture
    className="picture dark base"
  >
    <source
      media="(min-width: 1025px)"
      srcSet="https://www.aesop.com/medias/GelCleanser-640x640.png?context=bWFzdGVyfGltYWdlc3w1MzE5NTl8aW1hZ2UvcG5nfGltYWdlcy9oOTUvaGNkLzg4MDc3OTc4NTAxNDIucG5nfGM3NTNhYjdkMWNiOTRhNWZhNDA0NWExZWJiODMxN2VhNzEyY2ZkNGUyYzc4ZWVkZDRjYTIzMzZmZTRlOGYxNjM"
    />
    <source
      media="(min-width: 640px)"
      srcSet="https://www.aesop.com/medias/GelCleanser-640x640.png?context=bWFzdGVyfGltYWdlc3w1MzE5NTl8aW1hZ2UvcG5nfGltYWdlcy9oOTUvaGNkLzg4MDc3OTc4NTAxNDIucG5nfGM3NTNhYjdkMWNiOTRhNWZhNDA0NWExZWJiODMxN2VhNzEyY2ZkNGUyYzc4ZWVkZDRjYTIzMzZmZTRlOGYxNjM"
    />
    <source
      media="(min-width: 0px)"
      srcSet="https://www.aesop.com/medias/GelCleanser-640x640.png?context=bWFzdGVyfGltYWdlc3w1MzE5NTl8aW1hZ2UvcG5nfGltYWdlcy9oOTUvaGNkLzg4MDc3OTc4NTAxNDIucG5nfGM3NTNhYjdkMWNiOTRhNWZhNDA0NWExZWJiODMxN2VhNzEyY2ZkNGUyYzc4ZWVkZDRjYTIzMzZmZTRlOGYxNjM"
    />
    <img
      alt="Image of Milk Cleansing Campaign Aesop 2019"
    />
  </picture>
  <div
    className="contentWrapper"
  >
    <div
      className="content"
    >
      <article
        className="base dark undefined"
        id="ImageHeader-old-component"
      >
        <header
          className="base"
          id="ImageHeader-old-component"
        >
          <h2
            className="base xLarge dark heading heading"
          >
            Gentle Facial Cleansing Milk
          </h2>
        </header>
        <div
          className="copy dark"
        >
          Offering an exceptionally mild yet thorough cleanse, Gentle Facial Cleansing Milk removes surface impurities without stripping moisture, making it especially beneficial for dry or sensitive skin.
        </div>
        <div
          className="ctaWrapper"
        >
          <a
            className="base blockStyle hasIcon dark"
            href="http://google.com"
            target="_self"
          >
            Read more
            <i
              aria-hidden="true"
              className="icon"
            >
              <svg
                className="base rightArrow dark"
                data-ref=""
                focusable="false"
                height={15}
                role="img"
                tabIndex={-1}
                viewBox="0 0 50 50"
                width={15}
              >
                <g>
                  <path
                    className="undefined"
                    d="M30.1,5.3L50,25.1L30.1,45h-6.6l18-17.6H0v-4.8h41.5l-18-17.6h6.6V5.3z"
                  />
                </g>
              </svg>
            </i>
          </a>
        </div>
      </article>
    </div>
  </div>
</section>
`;
